README.md

```diff
@@ -37,4 +37,45 @@ pytest tests/test_scanner.py
 
 3. Click “Run 'pytest in tests'”
 
-4. Or click the green ▶ next to a test function
+4. Or click the green ▶️ next to a test function
+
+## Interpreter
+### Arithmetic operation rules
+```
+1. scalar + scalar -> standard arithmetic (e.g., 2 + 3 = 5)
+2. list + scalar  -> elementwise (e.g., [1, 2] + 3 = [4, 5])
+3. scalar + list  -> elementwise (e.g., 3 + [1, 2] = [4, 5])
+4. list + list    -> not supported (raises error)
+
+~ same rules apply for -, *, /
+~ if operands are not numbers or lists -> convert to a string and merge (in case of '+')
+```
+
+
+### Principles for List Handling
+
+#### Arithmetic and Comparison Work Elementwise
+If one or both arguments are lists, operations are applied element-by-element:
```

Comment:
This contradicts what you said earlier, where you said list + list isn't supported